packages/governance/src/governor/governor.cairo

```diff
@@ -0,0 +1,1041 @@
+        fn queue(
+            ref self: ComponentState<TContractState>, calls: Span<Call>, description_hash: felt252
+        ) -> felt252 {
+            let proposal_id = self._hash_proposal(calls, description_hash);
+            self.validate_state(proposal_id, array![ProposalState::Succeeded].span());
+
+            let eta_seconds = self.queue_operations(proposal_id, calls, description_hash);
+            assert(eta_seconds > 0, Errors::QUEUE_NOT_IMPLEMENTED);
+
+            let mut proposal = self.Governor_proposals.read(proposal_id);
+            proposal.eta_seconds = eta_seconds;
+            self.Governor_proposals.write(proposal_id, proposal);
+
+            self.emit(ProposalQueued { proposal_id, eta_seconds });
+
+            proposal_id
+        }
+
+        /// Executes a successful proposal. This requires the quorum to be reached, the vote to be
+        /// successful, and the deadline to be reached. Depending on the governor it might also be
+        /// required that the proposal was queued and that some delay passed.
+        ///
+        /// NOTE: Some modules can modify the requirements for execution, for example by adding an
+        /// additional timelock (See `timelock_controller`).
+        ///
+        /// Returns the id of the proposal.
+        ///
+        /// Requirements:
+        ///
+        /// - The proposal must be in the `Succeeded` or `Queued` state.
+        ///
+        /// Emits a `ProposalExecuted` event.
+        fn execute(
+            ref self: ComponentState<TContractState>, calls: Span<Call>, description_hash: felt252
+        ) -> felt252 {
+            let proposal_id = self._hash_proposal(calls, description_hash);
+            self
+                .validate_state(
+                    proposal_id, array![ProposalState::Succeeded, ProposalState::Queued].span()
+                );
+
+            // Mark proposal as executed to avoid reentrancy
+            let mut proposal = self.Governor_proposals.read(proposal_id);
+            proposal.executed = true;
+            self.Governor_proposals.write(proposal_id, proposal);
+
+            self.execute_operations(proposal_id, calls, description_hash);
+
+            self.emit(ProposalExecuted { proposal_id });
+
+            proposal_id
+        }
+
+        /// Cancels a proposal. A proposal is cancellable by the proposer, but only while it is
+        /// Pending state, i.e. before the vote starts.
+        ///
+        /// Returns the id of the proposal.
+        ///
+        /// Requirements:
+        ///
+        /// - The proposal must be in the `Pending` state.
+        ///
+        /// Emits a `ProposalCanceled` event.
+        fn cancel(
+            ref self: ComponentState<TContractState>, calls: Span<Call>, description_hash: felt252
+        ) -> felt252 {
+            let proposal_id = self._hash_proposal(calls, description_hash);
+            self.validate_state(proposal_id, array![ProposalState::Pending].span());
```

Comment:
Is it possible to cancel a queued proposal?

Reply:
Shouldn't be possible, why?